scripts/pipeline/orthology_benchmark.py

```diff
@@ -341,6 +341,39 @@ def extract_orthologs(res_dir: str, species_key1: str, species_key2: str) -> Lis
     return orthologs
 
 
+def extract_paralogs(res_dir: str, species_key: str) -> List[Tuple[str, str]]:
+    """Reads in putative paralogs inferred by OrthoFinder.
+
+    Args:
+        res_dir: Path to the directory with OrthoFinder results.
+        species_key: OrthoFinder identificator of the species of interest.
+
+    Returns:
+        A list of putative paralogous pairs for a specified species identificator used in the
+        OrthoFinder analysis.
+
+    """
+    paralogs = []
+
+    predictions_file = os.path.join(res_dir, "Gene_Duplication_Events", "Duplications.tsv")
+
+    with open(predictions_file) as file_handle:
+        reader = csv.DictReader(file_handle, delimiter="\t")
+        for row in reader:
+            species = row["Species Tree Node"]
+            genes1 = row["Genes 1"].split(", ")
+            genes2 = row["Genes 2"].split(", ")
+
+            if species == species_key:
+                for gene1 in genes1:
+                    for gene2 in genes2:
+                        paralogs.append(
+                            (gene1.split(f"{species_key}_")[1], gene2.split(f"{species_key}_")[1])
```

Comment:
Though this may well work here, using `split` could have unexpected results on different data. It might also cause confusion about what is being done here, since the species key is a prefix and not an internal delimiter of the kind that `split` is typically used for.

One way to remove the species-key prefix would be to define a `re.Pattern` object outside the for loops...
```python
prefix_pattern = re.compile(f"^{re.escape(species_key)}_")
```
...and then use its `sub` method on each `gene1` and `gene2` field to replace the pattern with an empty string. For example:
```python
gene1_bare_id = prefix_pattern.sub("", gene1)
```
Then you could append the tuple of bare IDs (e.g. `(gene1_bare_id, gene2_bare_id)`) to the `paralogs` list.

One disadvantage of that suggestion is that it won't raise an exception if the prefix is not found, while the current approach using `split` will complain in that case. If you want to raise an exception in the case that the species key isn't found, you could define a `re.Pattern` object to match the full species-gene pair...
```python
gene_pattern = re.compile(f'^{re.escape(species_key)}_(?P<gene_id>.+)$')
```
...and then take the bare gene ID from the match's capture group. For example:
```python
gene1_match = gene_pattern.match(gene1)
gene1_bare_id = gene1_match["gene_id"]
```
This will raise an exception if the prefix is not found in `gene1`.